cli/index.ts

```diff
@@ -206,8 +171,53 @@ const timeoutOption: INamedOption<yargs.Options> = {
   }
 };
 
-const defaultDatabaseOptionName = "default-database";
-const defaultLocationOptionName = "default-location";
+const defaultDatabaseOption: INamedOption<yargs.Options> = {
+  name: "default-database",
+  option: {
+    describe: "The default database to use. For BigQuery, this is a Google Cloud Project ID.",
+    type: "string"
+  },
+  check: (argv: yargs.Arguments<any>) => {
+    if (!argv[warehouseOption.name]) {
+      return;
+    }
+    if (
+      argv[defaultDatabaseOption.name] &&
+      !["bigquery", "snowflake"].includes(argv[warehouseOption.name])
+    ) {
+      throw new Error(
+        `The --${defaultDatabaseOption.name} flag is only used for BigQuery and Snowflake projects.`
+      );
+    }
+    if (!argv[defaultDatabaseOption.name] && argv[warehouseOption.name] === "bigquery") {
+      throw new Error(
+        `The --${defaultDatabaseOption.name} flag is required for BigQuery projects.`
+      );
+    }
```

Comment:
why are we requiring this? we should continue defaulting to whatever is in `dataform.json` (throughout)!

Reply:
`argv[warehouseOption.name]` isn't required for options other than `init`